modules/quantized-mesh/src/lib/parse-quantized-mesh.js

```diff
@@ -0,0 +1,69 @@
+import {getMeshBoundingBox} from '@loaders.gl/loader-utils';
+import decode, {DECODING_STEPS} from './decode';
+
+function getMeshAttributes(vertexData, header, bounds) {
+  const {minHeight, maxHeight} = header;
+  const [minX, minY, maxX, maxY] = bounds || [0, 0, 1, 1];
+  const xScale = maxX - minX;
+  const yScale = maxY - minY;
+  const zScale = maxHeight - minHeight;
+
+  const nCoords = vertexData.length / 3;
+  // vec3. x, y defined by bounds, z in meters
+  const positions = new Float32Array(nCoords * 3);
+
+  // vec2. 1 to 1 relationship with position. represents the uv on the texture image. 0,0 to 1,1.
+  const texCoords = new Float32Array(nCoords * 2);
+
+  // Data is not interleaved; all u, then all v, then all heights
+  for (let i = 0; i < nCoords; i++) {
+    const x = vertexData[i] / 32767;
+    positions[3 * i + 0] = x * xScale + minX;
+    texCoords[2 * i + 0] = x;
+  }
+
+  for (let i = 0; i < nCoords; i++) {
+    const y = vertexData[i + nCoords] / 32767;
+    positions[3 * i + 1] = y * yScale + minY;
+    texCoords[2 * i + 1] = y;
+  }
+
+  for (let i = 0; i < nCoords; i++) {
+    const z = vertexData[i + nCoords * 2] / 32767;
+    positions[3 * i + 2] = z * zScale + minHeight;
+  }
+
+  return {
+    POSITION: {value: positions, size: 3},
+    TEXCOORD_0: {value: texCoords, size: 2}
+    // NORMAL: {}, - optional, but creates the high poly look with lighting
+  };
+}
+
+function getTileMesh(arrayBuffer, options) {
+  if (!arrayBuffer) {
+    return null;
+  }
+  const {bounds} = options;
+  // Don't parse edge indices or format extensions
+  const {header, vertexData, triangleIndices} = decode(arrayBuffer, DECODING_STEPS.triangleIndices);
+  const attributes = getMeshAttributes(vertexData, header, bounds);
+
+  return {
+    // Data return by this loader implementation
+    loaderData: {
+      header: {}
+    },
+    header: {
+      vertexCount: triangleIndices.length,
+      boundingBox: getMeshBoundingBox(attributes)
+    },
+    mode: 4, // TRIANGLES
+    indices: {value: triangleIndices, size: 1},
+    attributes
+  };
+}
+
+export default async function loadQuantizedMesh(arrayBuffer, options) {
+  return getTileMesh(arrayBuffer, options.quantizedMesh);
```

Comment:
There's no `async` function inside `loadQuantizedMesh`, so there's no reason why `loadQuantizedMesh` has to be `async`, but I got an error without the `async` there.

Reply:
Oh I see, I need to separate `parse` and `parseSync`. Is there a `parseSync` as well as a `parseTextSync`?

https://github.com/visgl/loaders.gl/blob/9d45242b37a3d51d8f0e5f2fd4e2156b2121eade/modules/wkt/src/wkt-loader.js#L16-L17

Reply:
Hmm yes it is a little messy.

It should be enough to implement `either parseSync() or parseTextSync()`. If you usually parse directly from strings, it is of course more efficient not to have to go through array buffers.

Then, technically it would be enough to define `parseSync()`. But an idea was that if apps wanted to use the loaders directly (without the core module) we would make sure every loader exports a `parse()` function, so that apps could rely on that... But I am not sure direct loader calls are really used by any apps, though the tests will fail if there isn't a parse function.

Reply:
Well in this case it's a binary format, so `parseTextSync` doesn't make sense. But just implementing:
```
parse: async (arrayBuffer, options) => parseQuantizedMesh(arrayBuffer, options), 
parseSync: parseQuantizedMesh, 
```
seems helpful to give the `sync` option.

Reply:
Agreed - the sync option can make code a little easier to write and debug, which is why we offer it when the underlying parser supports it.